src/amo/components/AddonReview/index.js

```diff
@@ -61,16 +73,17 @@ export class AddonReviewBase extends React.Component<Props, State> {
     debounce: defaultDebounce,
   };
 
-  constructor(props: Props) {
+  constructor(props: InternalProps) {
     super(props);
     this.state = {
       reviewBody: props.review.body,
     };
+    invariant(props.createLocalState, 'props.createLocalState() was undefined');
     this.localState = props.createLocalState(`AddonReview:${props.review.id}`);
     this.checkForStoredState();
   }
 
-  componentWillReceiveProps(nextProps: Props) {
+  componentWillReceiveProps(nextProps: InternalProps) {
```

Comment:
Or maybe `Props` should be `BaseProps` and we could use `Props` for the set of all props.

Reply:
I think `Props` is nice because those are the actual component props that any calling code will need to care about. We could make it explicit like `ExternalProps` but, I dunno, I think it's implied.

Reply:
Alright, I am fine with that.